lightning/src/util/config.rs

```diff
@@ -315,6 +315,34 @@ impl Default for ChannelHandshakeLimits {
 	}
 }
 
+/// Options for how to set the max dust HTLC exposure allowed on a channel. See
+/// [`ChannelConfig::max_dust_htlc_exposure_msat`] for details.
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum MaxDustHTLCExposure {
+	/// This sets a fixed limit on the total dust exposure in millisatoshis. Setting this too low
+	/// may prevent the sending or receipt of low-value HTLCs on high-traffic nodes, and this limit
+	/// is very important to prevent stealing of dust HTLCs by miners.
+	///
+	/// Note that if the feerate increases significantly, without a manually increase
+	/// to this maximum the channel may be unable to send/receive HTLCs between the maximum dust
+	/// exposure and the new minimum value for HTLCs to be economically viable to claim.
+	FixedLimitMsat(u64),
+	/// This sets a multiplier on the estimated high priority feerate (sats/KW, as obtained from
+	/// [`FeeEstimator`]) to determine the maximum allowed dust exposure. If this variant is used
```

Comment:
I think here we should add a strong warning if you’re using a third-party estimator (i.e likely if you’re on mobile clients), due to this new `FeeRateMulitplier` you’re far more exposed to a fee griefing attack as the dust exposure can be inflated so high that you sent payment is unsafe.

Note this concern does apply to local fee estimator where the fees buckets are evaluated at each block tick like the Bitcoin Core one.

So I think we should add a warning here, or document in consequence our `FeeEstimator` ?

Reply:
Added docs on this enum

Reply:
Good enough comment, to be honest in term of fee-estimators a) there is a lot of trade-offs between them (e.g the Bitcoin Core is less reactive though more robust) and b) fee-estimation for Lightning is still rocket-science. The important nutshell is about lower-trust of third-party/outsourced fee-estimators for dust-exposure risks.